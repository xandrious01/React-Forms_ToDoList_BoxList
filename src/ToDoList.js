import React, { useState } from "react";
import Todo from './Todo';
import './ToDoList.css';
import NewTodoForm from "./NewTodoForm";
const { v4: uuidv4 } = require('uuid');


const ToDoList = () => {
    // const [editMode, setEditMode] = useState([false, {}]);
    
    const INITIAL_STATE = [];
    const [toDos, setToDos] = useState(INITIAL_STATE);

    const addToDo = (newTodo) => {
        return setToDos(toDos => [...toDos,
        { ...newTodo, id: uuidv4() }
        ]);
    };

    const remove = (id) => {
        return setToDos(toDos.filter(i => i.id !== id));
    }

    // const edit = (id, todo) => {
    //     return setEditMode([true, {
    //         id: id,
    //         todo: todo
    //     }]);
    // }

    // const renderInEditMode = () => {
    //     const [...obj] = editMode;
    //     console.log(obj)
    // toDos.map(i => {
    //     return i.id !== id ? i :
    //         (<input
    //             type="text"
    //             name="todo"
    //             id="todo"
    //             value={todo}
    //             onChange={addToDo}
    //         />)
    // })
    // setEditMode(true);
    // return renderInEditMode;

    return (
        <div>
            <h3>To Do List:</h3>
            <NewTodoForm addToDo={addToDo} />
            <ul>
                {toDos.map(todo =>
                (<Todo
                    key={todo.id}
                    id={todo.id}
                    // edit={edit}
                    remove={remove}
                    todo={todo.todo}
                />))}
            </ul>
        </div>
    )
}






export default ToDoList;