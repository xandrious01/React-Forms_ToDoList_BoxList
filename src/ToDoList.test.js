import React from "react";
import {screen, render, fireEvent} from '@testing-library/react'
import ToDoList from "./ToDoList";

it("renders without crashing", function() {
    render(<ToDoList />);
});

it("matches snapshot", function(){
    const {asFragment} = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new item", function(){
    const {queryByText, getByLabelText} = render(<ToDoList />);
    const input = getByLabelText("Enter a task:");
    const button = queryByText("Add!");
    expect(queryByText('Brush Teeth')).not.toBeInTheDocument();
    fireEvent.change(input, {target: { value: 'Brush Teeth'}})
    fireEvent.click(button);
    expect(queryByText('Brush Teeth')).toBeInTheDocument();
});

it("should render a working remove button with each task", function(){
    const {queryByText, getByLabelText} = render(<ToDoList />);
    const input = getByLabelText("Enter a task:");
    const button = queryByText("Add!");
    expect(queryByText('REMOVE')).not.toBeInTheDocument();
    fireEvent.change(input, {target : {value : 'walk dog'}});
    fireEvent.click(button);
    expect(queryByText('walk dog')).toBeInTheDocument();
    expect(queryByText('REMOVE')).toBeInTheDocument();
    const removeBtn = (queryByText("REMOVE"));
    fireEvent.click(removeBtn);
    expect(queryByText("walk dog")).not.toBeInTheDocument();
})