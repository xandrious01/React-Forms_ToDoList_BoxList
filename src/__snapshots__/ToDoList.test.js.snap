// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      To Do List:
    </h3>
    <form
      class="new-todo-form"
    >
      <label
        for="todo"
      >
        Enter a task:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder=""
        type="text"
        value=""
      />
      <button>
        Add!
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
